.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-added-large-files
+      - id: check-json
+      - id: check-toml
+      - id: check-merge-conflict
+      - id: debug-statements
+      - id: mixed-line-ending
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.14.6
```

Comment:
Good to see ruff updated to v0.14.6 (latest version). The configuration with `--fix` and `--exit-non-zero-on-fix` ensures automatic fixes while still failing the commit if issues remain.